udata_hydra/utils/csv.py

```diff
@@ -2,10 +2,13 @@
 
 
 async def detect_csv_from_headers(check) -> bool:
-    """Determine if content-type header looks like a csv's one"""
-    headers = json.loads(check["headers"] or {})
+    """
+    Determine if content-type header looks like a csv's one
+    or if it's binary for potential csv.gz
+    """
+    headers = json.loads(check["headers"] or "{}")
     return any(
         headers.get("content-type", "").lower().startswith(ct) for ct in [
             "application/csv", "text/plain", "text/csv"
         ]
-    )
+    ), headers.get("content-type", "").lower().startswith("application/octet-stream")
```

Comment:
Hmm this may end up downloading a lot of files. Can't we base our assumption on the file name in the URL instead? Not sure if any other hint could help?
cc @geoffreyaldebert

Reply:
Fixed in https://github.com/etalab/udata-hydra/pull/84/commits/3fa8dbafe557f0f971efb950d6234bf505085d25 : we have an additionnal check on the URL